docs/manuals/man1/pki-audit.1.md
# pki-audit 1 "Jun 30, 2015" PKI "PKI Audit Management Commands"

## NAME

pki-audit - Command-line interface for managing PKI audit configuration.

## SYNOPSIS

**pki** [*CLI-options*] **&lt;subsystem&gt;-audit**  
**pki** [*CLI-options*] **&lt;subsystem&gt;-audit-show** [*command-options*]  
**pki** [*CLI-options*] **&lt;subsystem&gt;-audit-mod** **--action** *action* [*command-options*]  
**pki** [*CLI-options*] **&lt;subsystem&gt;-audit-mod** **--input** *input-file* [*command-options*]  

## DESCRIPTION

The **pki-audit** commands provide command-line interfaces to manage audit configuration in the specified subsystem.
Currently the only supported subsystem is **tps**.

**pki** [*CLI-options*] **&lt;subsystem&gt;-audit**  
    This command is to list the available audit commands for the subsystem.

**pki** [*CLI-options*] **&lt;subsystem&gt;-audit-show** [*command-options*]  
    This command is to show the audit configuration in the subsystem.

**pki** [*CLI-options*] **&lt;subsystem&gt;-audit-mod** **--action** *action* [*command-options*]  
    This command is to change the audit (enabled/disabled) status in the subsystem.

**pki** [*CLI-options*] **&lt;subsystem&gt;-audit-mod** **--input** *input-file* [*command-options*]  
    This command is to modify the audit configuration in the subsystem.

## OPTIONS

The CLI options are described in **pki(1)**.

## OPERATIONS

To view available audit commands, type **pki &lt;subsystem&gt;-audit**.
To view each command's usage, type **pki &lt;subsystem&gt;-audit-&lt;command&gt; --help**.

All audit commands must be executed with the subsystem's admin authentication
(the user must be in the Administrators group).
See also the Authentication section in **pki(1)**.

### Viewing audit configuration

To view the audit configuration in TPS execute the following command:

```
$ pki <TPS admin authentication> tps-audit-show
```

To download the audit configuration from TPS into a file execute the following command:

```
$ pki <TPS admin authentication> tps-audit-show --output <output file>
```

### Changing audit status

To enable/disable audit in TPS, execute the following command:

```
$ pki <TPS admin authentication> tps-audit-mod --action <action>
```

where action is enable or disable.

### Modifying audit configuration

To modify the audit configuration in TPS, download the current configuration
using the above **tps-audit-show** command, edit the file,
then execute the following command:

```
$ pki <TPS admin authentication> tps-audit-mod --input <input file>
```

Optionally, a --output *output-file* option may be specified
to download the effective configuration after the modification.

## SEE ALSO

**pki(1)**

## AUTHORS

Endi S. Dewata &lt;edewata@redhat.com&gt;.

## COPYRIGHT

Copyright (c) 2015 Red Hat, Inc.
This is licensed under the GNU General Public License, version 2 (GPLv2).
A copy of this license is available at http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.


base/server/share/webapps/pki/js/pki-audit.js
/* --- BEGIN COPYRIGHT BLOCK ---
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; version 2 of the License.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along
 * with this program; if not, write to the Free Software Foundation, Inc.,
 * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Copyright (C) 2013 Red Hat, Inc.
 * All rights reserved.
 * --- END COPYRIGHT BLOCK ---
 *
 * @author Endi S. Dewata
 */

var AuditModel = Model.extend({
    initialize: function(attrs, options) {
        options = options || {};
        var self = this;
        AuditModel.__super__.initialize.call(self, attrs, options);
        self.url = options.url;
    },
    parseResponse: function(response) {
        return {
            id: "audit",
            status: response.Status,
            signed: response.Signed,
            interval: response.Interval,
            bufferSize: response.BufferSize,
            events: response.Events.Event
        };
    },
    createRequest: function(entry) {
        return {
            Status: entry.status,
            Signed: entry.signed,
            Interval: entry.interval,
            BufferSize: entry.bufferSize,
            Events: {
                Event: entry.events
            }
        };
    },
    changeStatus: function(action, options) {
        var self = this;
        var url;
        if (typeof self.url == 'function') {
            url = self.url();
        } else {
            url = self.url;
        }
        $.ajax({
            type: "POST",
            url: url + "?action=" + action,
            dataType: "json"
        }).done(function(data, textStatus, jqXHR) {
            self.set(self.parseResponse(data));
            if (options.success) options.success.call(self, data, textStatus, jqXHR);
        }).fail(function(jqXHR, textStatus, errorThrown) {
            if (options.error) options.error.call(self, jqXHR, textStatus, errorThrown);
        });
    }
});

var AuditTableItem = TableItem.extend({
    initialize: function(options) {
        var self = this;
        AuditTableItem.__super__.initialize.call(self, options);
    },
    renderColumn: function(td, templateTD) {
        var self = this;

        AuditTableItem.__super__.renderColumn.call(self, td, templateTD);

        $("a", td).click(function(e) {
            e.preventDefault();
            self.open();
        });
    },
    open: function() {
        var self = this;

        var value = self.get("value");
        var dialog;

        if (self.table.mode == "view" || value == "mandatory") {
            // In view mode all events are read-only.
            // Mandatory events are always read-only.
            dialog = new Dialog({
                el: self.table.parent.$("#event-dialog"),
                title: "Event",
                readonly: ["name", "value"],
                actions: ["close"]
            });

        } else if (self.table.mode == "edit" && value != "mandatory") {
            // Optional events are editable in edit mode.
            dialog = new Dialog({
                el: self.table.parent.$("#event-dialog"),
                title: "Edit Event",
                readonly: ["name"],
                actions: ["cancel", "save"]
            });

            dialog.handler("save", function() {

                // save changes
                dialog.save();
                _.extend(self.entry, dialog.entry);

                // redraw table
                self.table.render();
                dialog.close();
            });
        }

        dialog.entry = _.clone(self.entry);

        dialog.open();
    }
});

var AuditPage = EntryPage.extend({
    initialize: function(options) {
        var self = this;
        AuditPage.__super__.initialize.call(self, options);
    },
    setup: function() {
        var self = this;

        AuditPage.__super__.setup.call(self);

        self.enableAction = $("[name='enable']", self.viewMenu);
        self.disableAction = $("[name='disable']", self.viewMenu);

        $("a", self.enableAction).click(function(e) {
            e.preventDefault();
            self.changeStatus("enable", "Are you sure you want to enable this entry?");
        });

        $("a", self.disableAction).click(function(e) {
            e.preventDefault();
            self.changeStatus("disable", "Are you sure you want to disable this entry?");
        });

        self.eventsTable = new Table({
            el: self.$("table[name='events']"),
            columnMappings: {
                id: "name"
            },
            pageSize: 10,
            tableItem: AuditTableItem,
            parent: self
        });
    },
    renderContent: function() {
        var self = this;

        AuditPage.__super__.renderContent.call(self);

        var status = self.entry.status;
        if (status == "Disabled") {
            self.editAction.show();
            self.enableAction.show();
            self.disableAction.hide();

        } else {
            self.editAction.hide();
            self.enableAction.hide();
            self.disableAction.show();
        }

        if (self.mode == "edit") {
            self.eventsTable.mode = "edit";

        } else { // self.mode == "view"
            self.eventsTable.mode = "view";
        }

        self.eventsTable.entries = self.entry.events;
        self.eventsTable.render();
    },
    saveFields: function() {
        var self = this;

        AuditPage.__super__.saveFields.call(self);

        self.entry.events = self.eventsTable.entries;
    }
});
